### DISCUSSION TITLE
Reading truncated text in cells

### DISCUSSION THREAD

**Discussion by paskn** [saulpw/visidata#1938] [Q&A] [answered]
Hi! I am learning VisiData, and it is amazing. Thank you for this tool!

I am exploring how can I use VisiData for a workflow where I have to read text in cells. Basically, is it possible to quickly expand truncated text to read it in full? I struggle finding info on truncation of long strings and whether I can do anything about it.

Thanks!

**Comment by ajkerrigan** [comment, marked as answer]
There are a few different ways to expand truncated text. Some common ones:

- `resize-col-max` (`_`): Expand the column to fit the text (up to the width of your terminal). Useful for some cases.
- `visibility-sheet` (`v`): Let each row of data take up multiple display rows, so long text can wrap.
- `open-cell` (`zEnter`): Open a new sheet with the contents of a cell.

This shows the effects of those commands on the [benchmark.csv sample data](https://github.com/saulpw/visidata/blob/2b3a569772d734278ba1fbecc9da2db6f88abf24/sample_data/benchmark.csv) with default column widths:

[![asciicast](https://asciinema.org/a/593618.svg)](https://asciinema.org/a/593618)

**Reply by saulpw** [reply]
Great summary, @ajkerrigan.  I would also add `e Ctrl+O`, which pulls the cell contents into $EDITOR.  (Any saved changes will be applied to the sheet, however.)

**Reply by paskn** [reply]
Thank you!
